django_ontruck/monads/result.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import annotations
+
+from abc import ABCMeta, abstractmethod
+from typing import Callable, Optional, TypeVar, Generic, Union
+
+
+ValueType = TypeVar('ValueType')
+U = TypeVar('U')  # generic type for values
+
+ErrorType = TypeVar('ErrorType')
+F = TypeVar('F')  # generic type for errors
+
+
+class Result(Generic[ValueType, ErrorType], metaclass=ABCMeta):
+    """Result monad interface.
+    """
+    _inner_value: Union[ValueType, ErrorType]
+
+    @abstractmethod
+    def is_ok(self) -> bool:
```

Comment:
Would it make sense to alias this as `__bool__`?

Reply:
totally, I'll add it